pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -48,39 +48,44 @@ func (d *DualWriterMode2) Mode() DualWriterMode {
 }
 
 // Create overrides the behavior of the generic DualWriter and writes to LegacyStorage and Storage.
-func (d *DualWriterMode2) Create(ctx context.Context, original runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+func (d *DualWriterMode2) Create(ctx context.Context, in runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	var method = "create"
 	log := d.Log.WithValues("method", method)
 	ctx = klog.NewContext(ctx, log)
 
 	startLegacy := time.Now()
-	created, err := d.Legacy.Create(ctx, original, createValidation, options)
+	createdFromLegacy, err := d.Legacy.Create(ctx, in, createValidation, options)
 	if err != nil {
 		log.Error(err, "unable to create object in legacy storage")
 		d.recordLegacyDuration(true, mode2Str, d.resource, method, startLegacy)
-		return created, err
+		return createdFromLegacy, err
 	}
 	d.recordLegacyDuration(false, mode2Str, d.resource, method, startLegacy)
 
-	if err := enrichLegacyObject(original, created); err != nil {
-		return created, err
+	accIn, err := meta.Accessor(in)
```

Comment:
i think this check could be removed. the underlying storage layer should validate this

Reply:
I'd rather keep returning the error here as it's easier for someone without the whole context or know from where this is called to understand